frame/bounties/src/lib.rs

```diff
@@ -653,12 +777,608 @@ decl_module! {
+										}
+									}
+								},
+							}
+						},
+						SubBountyStatus::PendingPayout { ref subcurator, .. } => {
+							ensure!(
+								maybe_sender.map_or(
+									true,
+									|sender| sender == master_curator,
+								),
+								BadOrigin,
+							);
+							slash_curator(subcurator, &mut subbounty.curator_deposit);
+							// Continue to change bounty status below...
+						},
+					};
+					// Move the subbounty state to Added.
+					subbounty.status = SubBountyStatus::Added;
+					Ok(())
+				}
+			)?;
+		}
+
+		/// Award subbounty to a beneficiary.
+		///
+		/// The beneficiary will be able to claim the
+		/// funds after a delay.
+		///
+		/// The dispatch origin for this call must be
+		/// the master curator or subcurator of this subbounty.
+		///
+		/// Parent bounty must be in active state,
+		/// for this subbounty call to work.
+		///
+		/// Subbounty must be in active state, for
+		/// processing the call. and state of subbounty is
+		/// moved to PendingPayout on successful call
+		/// completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		/// - `beneficiary`: Beneficiary account.
+		#[weight = <T as Config>::WeightInfo::award_subbounty()]
+		fn award_subbounty(origin,
+			#[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+			beneficiary: <T::Lookup as StaticLookup>::Source
+		) {
+			let signer = ensure_signed(origin)?;
+			let beneficiary = T::Lookup::lookup(beneficiary)?;
+
+			// Ensure parent bounty is Active
+			let (_, _) = Self::ensure_bounty_active(bounty_id)?;
+
+			// Ensure subbounty is in expected state
+			SubBounties::<T>::try_mutate_exists(
+				bounty_id,
+				subbounty_id,
+				|maybe_subbounty| -> DispatchResult {
+					let mut subbounty = maybe_subbounty
+						.as_mut()
+						.ok_or(Error::<T>::InvalidIndex)?;
+
+					// Ensure Subbounty is in active state
+					if let SubBountyStatus::Active { ref subcurator } =
+						subbounty.status
+					{
+						ensure!(
+							signer == *subcurator,
+							Error::<T>::RequireSubCurator,
+						);
+						// Move the subbounty state to Pending payout.
+						subbounty.status = SubBountyStatus::PendingPayout {
+							subcurator: signer,
+							beneficiary: beneficiary.clone(),
+							unlock_at: system::Module::<T>::block_number() +
+								T::BountyDepositPayoutDelay::get(),
+						};
+						Ok(())
+					} else {
+						Err(Error::<T>::UnexpectedStatus.into())
+					}
+				}
+			)?;
+			// Trigger the event SubBountyAwarded
+			Self::deposit_event(
+				Event::<T>::SubBountyAwarded(
+					bounty_id,
+					subbounty_id,
+					beneficiary
+				)
+			);
+		}
+
+		/// Claim the payout from an awarded subbounty after payout delay.
+		///
+		/// The dispatch origin for this call may be any signed origin.
+		///
+		/// Call works independent of parent bounty state,
+		/// No need for parent bounty must be in active state.
```

Comment:
there is no need, but AFAICT bounty cannot be awarded if there is pending subbounty. And subbounty only gets removed in claim_award. So bounty **is** in active state anyway.

Reply:
ok now I see, the bounty can be in Active state or Funded state or CuratorProposed state.